src/main/java/com/flipsmart/FlipAssistOverlay.java

```diff
@@ -0,0 +1,784 @@
+		CONFIRM_OFFER("Confirm", "Click the confirm button"),
+		WAITING_BUY("Buying...", "Offer placed, waiting to fill"),
+		COLLECT_ITEMS("Collect", "Items bought! Collect from slot"),
+		SELL_ITEMS("Sell Items", "Click SELL on collected items"),
+		SET_SELL_PRICE("Set Sell Price", "Press [%s] to set: %s"),
+		CONFIRM_SELL("Confirm Sale", "Click confirm to list"),
+		WAITING_SELL("Selling...", "Listed! Waiting to sell"),
+		COMPLETE("Complete!", "Flip finished - collect GP");
+		
+		private final String title;
+		private final String description;
+		
+		FlipAssistStep(String title, String description)
+		{
+			this.title = title;
+			this.description = description;
+		}
+		
+		public String getTitle()
+		{
+			return title;
+		}
+		
+		public String getDescription()
+		{
+			return description;
+		}
+	}
+	
+	@Getter
+	private FlipAssistStep currentStep = FlipAssistStep.SELECT_ITEM;
+	
+	@Inject
+	private FlipAssistOverlay(Client client, ClientThread clientThread, FlipSmartConfig config, ItemManager itemManager)
+	{
+		this.client = client;
+		this.clientThread = clientThread;
+		this.config = config;
+		this.itemManager = itemManager;
+		
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		setPriority(PRIORITY_HIGH);
+		setMovable(true);
+	}
+	
+	public void updateStep()
+	{
+		if (focusedFlip == null)
+		{
+			currentStep = FlipAssistStep.SELECT_ITEM;
+			return;
+		}
+		
+		if (!isGrandExchangeOpen())
+		{
+			currentStep = focusedFlip.isBuying() ? FlipAssistStep.WAITING_BUY : FlipAssistStep.WAITING_SELL;
+			return;
+		}
+		
+		int inputType = getInputType();
+		if (inputType == INPUT_TYPE_GE_SEARCH)
+		{
+			currentStep = FlipAssistStep.SEARCH_ITEM;
+			return;
+		}
+		
+		if (inputType == INPUT_TYPE_NUMERIC)
+		{
+			currentStep = determineNumericInputStep();
+			return;
+		}
+		
+		if (isOfferSetupOpen())
+		{
+			currentStep = determineOfferSetupStep();
+			return;
+		}
+		
+		currentStep = focusedFlip.isBuying() ? FlipAssistStep.SELECT_ITEM : FlipAssistStep.SELL_ITEMS;
+	}
+	
+	private FlipAssistStep determineNumericInputStep()
+	{
+		if (isLikelyPriceInput())
+		{
+			return focusedFlip.isBuying() ? FlipAssistStep.SET_PRICE : FlipAssistStep.SET_SELL_PRICE;
+		}
+		return FlipAssistStep.SET_QUANTITY;
+	}
+	
+	private FlipAssistStep determineOfferSetupStep()
+	{
+		boolean qtyCorrect = isValueWithinTolerance(getCurrentQuantityFromGE(), focusedFlip.getCurrentStepQuantity());
+		boolean priceCorrect = isValueWithinTolerance(getCurrentPriceFromGE(), focusedFlip.getCurrentStepPrice());
+		
+		if (qtyCorrect && priceCorrect)
+		{
+			return focusedFlip.isBuying() ? FlipAssistStep.CONFIRM_OFFER : FlipAssistStep.CONFIRM_SELL;
+		}
+		if (!qtyCorrect)
+		{
+			return FlipAssistStep.SET_QUANTITY;
+		}
+		return focusedFlip.isBuying() ? FlipAssistStep.SET_PRICE : FlipAssistStep.SET_SELL_PRICE;
+	}
+	
+	private boolean isValueWithinTolerance(int current, int target)
+	{
```

Comment:
Might be worth adding a docstring/comment here explaining why we need the tolerance check